File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
 
public class Solution {
    public List<List<int>> LevelOrder(TreeNode root) {
        List<List<int>> levels = new List<List<int>>();
        Queue<TreeNode> queue = new Queue<TreeNode>();

       if(root != null) {
        queue.Enqueue(root);
       }
        // levels.Add(new List<int>{root.val})
        while(queue.Count > 0){
            List<int> level = new List<int>();
            for(int i = 0, len = queue.Count; i < len; i++){
                TreeNode curr = queue.Dequeue();
                level.Add(curr.val);
                if(curr.left != null){
                    queue.Enqueue(curr.left);
                }
                if(curr.right != null){
                    queue.Enqueue(curr.right);
                }
            }
            levels.Add(level);
        }
        return levels;
    }
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {TreeNode} subRoot
     * @return {boolean}
     */
    isSubtree(root, subRoot) {
        if(subRoot == null) {
            return true;
        }
        if(root == null) {
            return false;
        }
        if(this.isSameTree(root, subRoot)){
            return true;
        }
        return this.isSubtree(root.left, subRoot) || this.isSubtree(root.right, subRoot);
    }
    isSameTree(tree1, tree2){
        if(tree1 == null || tree2 == null){
            return tree1 == tree2;
        }
        if(tree1.val != tree2.val){
            return false;
        }
        return this.isSameTree(tree1.left, tree2.left) && this.isSameTree(tree1.right, tree2.right);
    }
}
